### ISSUE TITLE
OCLint does not report compilation failures using PMD Reporter

### ISSUE THREAD

**Issue by Uranium-92** [oclint/oclint#398]
### Issue Summary
If OCLint fails to compile source file during analysis using PMD Reporter, the resulting analysis report does not have any signs of that, nor analysis is failed.
However OCLint reports compilation errors when is run with Plain Text Report.

### Environment
- Operation system name and version: Mac OS X 10.11.6
- OCLint version: 0.11.1
- How OCLint is installed: binary downloaded from GitHubvim

### Reproduction Steps
Run analysis using PMD Reporter against source file with compilation error; or simply pass non-existent warning-flag as compilation argument along with `-Werror`.

#### Sample code

File main.cpp:
```
#include <iostream>

int main(int args, char* argv[])
{
	std::cout << "Hello, World!" << std::endl;
}
```

Command to run OCLint with the code above:
`oclint --report-type pmd -o result.xml main.cpp -- -Wnon-exitent-warning -Werror`

### Expected Behavior
The resulting analysis report file result.xml has entries about compilation errors, or analysis is failed with descriptive message printed to stderr.

### Actual Behavior
The resulting analysis report file result.xml does not have any signs of compilation errors, nor analysis is failed.

**Comment by Uranium-92** [comment]
It seems that for consistency OCLint must add entries to resulting report file but the whole compilation should not be failed.
@ryuichis, what do you think? Can make pull request if you approve this behavior.

**Comment by ryuichis** [comment]
@Uranium-92 thanks for the info, and please go ahead and proceed with what you have in your mind. I also checked on the usage on the pmd report, and it appears to be the top choice for OCLint users, and almost triple the second place (text report). So it will be beneficial for a big majority of users.
